src/Azure.Functions.Cli/Helpers/NetworkHelpers.cs

```diff
@@ -33,5 +37,33 @@ public static int GetAvailablePort()
                 listener.Stop();
             }
         }
+
+        public static int GetNextAvailablePort(int startPort)
+        {
+            var usedPorts = new List<int>();
+            var ipGlobalProperties = IPGlobalProperties.GetIPGlobalProperties();
+
+            usedPorts.AddRange(ipGlobalProperties
+                                .GetActiveTcpConnections()
+                                .Where(c => c.LocalEndPoint.Port >= startPort)
+                                .Select(c => c.LocalEndPoint.Port));
+            usedPorts.AddRange(ipGlobalProperties
+                                .GetActiveTcpListeners()
+                                .Where(l => l.Port >= startPort)
+                                .Select(l => l.Port));
+
+            usedPorts.Sort();
+
+            foreach (var usedPort in usedPorts)
+            {
+                if (startPort < usedPort)
+                {
+                    return startPort;
+                }
+                startPort++;
+            }
+
```

Comment:
What if all ports are exhausted, then should we log the errors?

Reply:
After checking the ports after the start port, the `GetAvailablePort` method will try to get any available port at random.